src/variable.cpp

```diff
@@ -77,6 +90,37 @@ void Variable::swap(Variable &rhs)
     std::swap(this->mPimpl, rhs.mPimpl);
 }
 
+void Variable::addEquivalence(const VariablePtr &variable1, const VariablePtr &variable2)
+{
+    variable1->setEquivalentTo(variable2);
+    variable2->setEquivalentTo(variable1);
+}
+
+VariablePtr Variable::getEquivalentVariable(size_t index)
+{
+    VariableWeakPtr weakEquivalentVariable = mPimpl->mEquivalentVariables.at(index);
+    return weakEquivalentVariable.lock();
+}
+
+size_t Variable::equivalentVariableCount() const
+{
+    return mPimpl->mEquivalentVariables.size();
+}
+
+bool Variable::hasEquivalentVariable(const VariablePtr &equivalentVariable)
+{
+    auto result = mPimpl->findEquivalentVariable(equivalentVariable);
+    return result != mPimpl->mEquivalentVariables.end();
```

Comment:
I've swapped in @agarny 's suggestion but you are right, this does violate the Google style guide on [max line lengths of 80 characters](https://google.github.io/styleguide/cppguide.html#Line_Length). We do appear to be +80 characters in many places though so it might be worth deciding if we want to rigidly follow Google on this or add an exception to our coding style.